plume-api/src/lib.rs

```diff
@@ -0,0 +1,18 @@
+extern crate canapi;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+
+macro_rules! api {
+    ($url:expr => $ep:ty) => {
+        impl Endpoint for $ep {
+            type Id = i32;
+
+            fn endpoint() -> &'static str {
+                $url
+            }
+        }
+    };
+}
```

Comment:
sometimes i dream of lisp indentation style in Javascript. soon, I'll start dreaming of lisp indentation style in rust